staging/src/k8s.io/code-generator/cmd/import-boss/README.md

```diff
@@ -0,0 +1,55 @@
+## Purpose
+
+- `import-boss` enforces import restrictions in a given repository.
+
+## How does it work?
+
+- When a directory is verified, `import-boss` looks for a file called `.import-restrictions`. If this file is not found, this directory will be skipped.
+
+- Adding `.import-restrictions` files does not add them to CI runs. They need to be explicitly added to `hack/verify-import-boss.sh`.
+
+- If an `.import-restrictions` file is found, then all imports of the package are checked against each rule in the file. A rule consists of three parts:
+  - A `SelectorRegexp`, to select the import paths that the rule applies to.
+  - A list of `AllowedPrefixes`
+  - A list of `ForbiddenPrefixes`
+
+- An import is allowed if it matches at least one allowed prefix and does not match any forbidden prefixes. An example `.import-restrictions` file looks like this:
+
+```json
+{
+  "Rules": [
+    {
+      "SelectorRegexp": "k8s[.]io",
+      "AllowedPrefixes": [
+        "k8s.io/gengo/examples",
+        "k8s.io/kubernetes/third_party"
+      ],
+      "ForbiddenPrefixes": [
+        "k8s.io/kubernetes/pkg/third_party/deprecated"
+      ]
+    },
+    {
+      "SelectorRegexp": "^unsafe$",
+      "AllowedPrefixes": [
+      ],
+      "ForbiddenPrefixes": [
+        ""
+      ]
+    }
+  ]
+}
+```
+- Take note of `"SelectorRegexp": "k8s[.]io"` in the first block. This specifies that we are applying these rules to the `"k8s.io"` import path.
+- The second block explicitly matches the "unsafe" package, and forbids it ("" is a prefix of everything).
+
+## How do I run import-boss?
```

Comment:
this section is a little odd here because it only applies to the k/k repo. At least mention that.

background: this file will show up at https://github.com/kubernetes/code-generator/tree/master/cmd/import-boss too.

Reply:
Something like this @sttts ?

`## How do I run import-boss within the k/k repo?`

Reply:
Yes.